addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -3513,6 +3525,137 @@ QUnit.test('receive new needaction messages', async function (assert) {
+        // mocked models don't have computed fields
+        needaction: true,
+        model: 'res.partner',
+        record_name: 'Refactoring',
+        res_id: 20,
+    });
+    this.data['mail.notification'].records.push({
+        mail_message_id: 100, // id of related message
+        res_partner_id: this.data.currentPartnerId, // must be for current partner
+    });
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.box_inbox',
+            },
+        },
+    });
+    assert.containsNone(
+        document.body,
+        '.o_Message_commandUnfollow',
+        "should not have button unfollow"
+    );
+});
+
+QUnit.test('should display unfollow button when current user is follower of the thread', async function (assert) {
+    assert.expect(1);
+
+    this.data['res.partner'].records.push({
+        id: 20,
+        message_follower_ids: [1],
+    });
+    this.data['mail.followers'].records.push({
+        id: 1,
+        partner_id: this.data.currentPartnerId,
+        res_id: 20,
+        res_model: 'res.partner',
+    });
+    this.data['mail.message'].records.push({
+        body: "<p>Test</p>",
+        id: 100,
+        needaction: true,
+        model: 'res.partner',
+        record_name: 'Refactoring',
+        res_id: 20,
+    });
+    this.data['mail.notification'].records.push({
+        mail_message_id: 100,
+        res_partner_id: this.data.currentPartnerId,
+    });
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.box_inbox',
+            },
+        },
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_Message_commandUnfollow',
+        "should have button unfollow"
+    );
+});
+
+QUnit.test('unfollow thread from inbox', async function (assert) {
+    assert.expect(3);
+
+    this.data['res.partner'].records.push({
+        id: 20,
+        message_follower_ids: [1],
+    });
+    this.data['mail.followers'].records.push({
+        id: 1,
+        partner_id: this.data.currentPartnerId,
+        res_id: 20,
+        res_model: 'res.partner',
+    });
+    // message that is expected to be found in Inbox
+    this.data['mail.message'].records.push({
+        body: "<p>Test</p>",
+        id: 100,
+        needaction: true,
+        model: 'res.partner',
+        record_name: 'Refactoring',
+        res_id: 20,
+    });
+    this.data['mail.notification'].records.push({
+        mail_message_id: 100,
+        res_partner_id: this.data.currentPartnerId,
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_post') {
+                assert.step('message_post');
+            }
+            if (route.includes('message_unsubscribe')) {
+                assert.step('rpc:message_unsubscribe');
+            }
+            return this._super(...arguments);
+        },
```

Comment:
Avoid using `mockRPC`. You can replace it by:
1. ideally, check the result of the rpc in the interface if possible
2. if not possible, make a read rpc to verify the server value that was supposed to have changed (this requires the mock rpc to be properly implemented for this feature).